org.eclipse.jgit/src/org/eclipse/jgit/diff/DiffFormatter.java

```diff
@@ -651,6 +657,48 @@ public void format(List<? extends DiffEntry> entries) throws IOException {
 			format(ent);
 	}
 
+	/**
+	 * Format the diff entries by filtering out the noise from the given delta filter pattern
+	 * The filter acts only for files that have MODIFY change Type.
+	 * If there are no changes detected, we will remove the diff entry.
+	 * @param entries
+	 * @param deltaFilterPattern
+	 * @throws IOException
+	 */
+
+	public void format(List<? extends DiffEntry> entries, Pattern deltaFilterPattern) throws IOException {
+		if (deltaFilterPattern == null)
+			format(entries);
```

Comment:
Also empty string ("")? also make a void return after this and we can eliminate the else block.

Reply:
deltaFilterPattern is an object. And if the pattern is null, it goes to the usual Jgit flow which is the format(entries).

Reply:
Ok, then just do the `return;` after so we can eliminate the else block.

Reply:
changed it